### PR TITLE
azure-arm-template: get CustomScriptExtension output

### PR DESCRIPTION
**Pull request by dennisameling**
Used [this](https://github.com/azure/azure-quickstart-templates/tree/master/quickstarts/microsoft.compute/vm-custom-script-output) for inspiration.

### CODE REVIEW

**Comment by dennisameling** [comment]
I was able to run the ARM template directly against Azure, so that part is working correctly. Now we only have to make sure that GitHub outputs the logs correctly, but [that should be relatively straightforward](https://github.com/marketplace/actions/deploy-azure-resource-manager-arm-template#steps-1). Might have some time tomorrow to look into that.

**Comment by dennisameling** [comment]
Ok, [let's try if this works](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/3920543615/jobs/6702240114) (I'll remove the VM right after creation).

**Comment by dennisameling** [comment]
That [worked](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/3920543615/jobs/6702240114#step:8:1)! 🎉 

We might want to do some basic JSON parsing though to make things easier to read. Will follow up on that later.

CC @dscho

File: azure-self-hosted-runners/azure-arm-template.json [hunk 1]
```diff
@@ -108,7 +108,12 @@
         "nicName": "[concat(parameters('virtualMachineName'), '-nic')]",
         "vnetName": "[concat(parameters('virtualMachineName'), '-vnet')]",
         "vnetId": "[resourceId(resourceGroup().name,'Microsoft.Network/virtualNetworks', concat(parameters('virtualMachineName'), '-vnet'))]",
-        "subnetRef": "[concat(variables('vnetId'), '/subnets/', parameters('subnetName'))]"
+        "subnetRef": "[concat(variables('vnetId'), '/subnets/', parameters('subnetName'))]",
+        "UriFileNamePieces": "[split(parameters('postDeploymentPsScriptUrl'), '/')]",
+        "firstFileNameString": "[variables('UriFileNamePieces')[sub(length(variables('UriFileNamePieces')), 1)]]",
+        "firstFileNameBreakString": "[split(variables('firstFileNameString'), '?')]",
+        "firstFileName": "[variables('firstFileNameBreakString')[0]]",
```
Comment (dennisameling) [inline azure-self-hosted-runners/azure-arm-template.json:115]: Taken directly from the [original template we used earlier](https://catalogartifact.azureedge.net/publicartifacts/Microsoft.CustomScriptExtension-arm-2.0.57/MainTemplate.json).

**Comment by dennisameling** [comment]
It [works](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/3923027664/jobs/6706361186#step:8:17)!! 🎉 

(Have just deleted that VM too)

**Comment by dscho** [comment]
> It [works](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/3923027664/jobs/6706361186#step:8:17)!! 🎉
> 
> (Have just deleted that VM too)

Exciting! ❤️

**Review by dscho — APPROVED** [review]
Wonderful work, thank you very much!

**Pull request merged**